bigquery/test/datasets.test.js

```diff
@@ -0,0 +1,233 @@
+// Copyright 2016, Google, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+var proxyquire = require('proxyquire').noCallThru();
+var datasetId = 'foo';
+var projectId = process.env.GCLOUD_PROJECT;
+
+function getSample () {
+  var tableMock = {
+    get: sinon.stub(),
+    metadata: {
+      numBytes: 1000000
+    }
+  };
+  tableMock.get.callsArgWith(0, null, tableMock);
+  var tablesMock = [
+    tableMock
+  ];
+  var datasetsMock = [
+    {
+      id: datasetId
```

Comment:
Can this go into 3 lines, e.g.
`var datasetsMock = [{`
`   id: datasetId`
`}];`